src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexParseException.cs

```diff
@@ -5,23 +5,47 @@
 
 namespace System.Text.RegularExpressions
 {
+    /// <summary>
+    /// An exception as a result of a parse error in a regular expression <see cref="RegularExpressions"/>, with
+    /// detailed information in the <see cref="Error"/> and <see cref="Offset"/> properties.
+    /// </summary>
     [Serializable]
-    internal sealed class RegexParseException : ArgumentException
+    public sealed class RegexParseException : ArgumentException
     {
-        private readonly RegexParseError _error; // tests access this via private reflection
-
         /// <summary>Gets the error that happened during parsing.</summary>
-        public RegexParseError Error => _error;
+        public RegexParseError Error { get; }
 
         /// <summary>Gets the offset in the supplied pattern.</summary>
         public int Offset { get; }
 
-        public RegexParseException(RegexParseError error, int offset, string message) : base(message)
+        internal RegexParseException(RegexParseError error, int offset, string message) : base(message)
+        {
+            Error = error;
+            Offset = offset;
+        }
+
+        /// <summary>
+        /// Construct a custom RegexParseException that creates a default message based on the given <see cref="RegexParseError"/> value.
+        /// </summary>
+        /// <param name="error">The <see cref="RegexParseError"/> value detailing the type of parse error.</param>
+        /// <param name="offset">The zero-based offset in the regular expression where the parse error occurs.</param>
+        public RegexParseException(RegexParseError error, int offset) : base(MakeMessage(error, offset))
         {
-            _error = error;
+            Error = error;
             Offset = offset;
         }
 
+        private static string MakeMessage(RegexParseError error, int offset)
+        {
+            return SR.Format(SR.Unknown, error.ToString(), offset);
```

Comment:
I don't understand this.  The resx appears to have custom strings for all of the error values.  Does it not?  If it does, why aren't we using them?

Reply:
This is by specific request, see earlier discussion. There's no pattern here, and so the other strings cannot be used, which require a pattern, offset and sometimes more data, so we cannot use that, and we rolled that back. This was agreed with the design team and above with @danmosemsft.

Reply:
If there's insufficient data to appropriately construct the exception, then why is this ctor valuable?

Also, many/most of the error messages do not require additional data.

Reply:
> Also, many/most of the error messages do not require additional data.

1. Each requires the `pattern` as a string. Which is not available. 
2. There was a strong request to add `Unknown` for future situations where a new type of error would be found, but without the wish to change the public api. `Unknown` could be used, possibly with a different string in that case.
3. There was a strong wish during api review for a "default" of the enum: the zero value should not be a specific error.
4. The Unknown string is somewhat redundant, sure, but it serves well for the exception ctor, which was also requested/agreed by the api review board. I wished for more control over the message and such, and passing in the `pattern`, but that was denied in favor of a simple default string.

And don't think we should try to go back to the drawing board. It serves a niche purpose: users _can_ create a custom exception if they need to, but per how `ArgumentException` works, cannot override the `Message` and we cannot give them a useful message by lack of information. It was also explicitly denied to make this _non-sealed_, which imho would've been a better approach for end users. The reason: compat with binary serialization.

Reply:
> Each requires the pattern as a string. Which is not available.

Not true. For example, AlternationHasComment has no holes.

I'm saying we should use the descriptive message at least in addition to if not instead of the default when we can.